src/modules/battery.cpp

```diff
@@ -156,6 +174,47 @@ const std::tuple<uint8_t, float, std::string, float> waybar::modules::Battery::g
   std::lock_guard<std::mutex> guard(battery_list_mutex_);
 
   try {
+#if defined(__FreeBSD__)
+    int capacity;
+    size_t size_capacity = sizeof capacity;
+    if (sysctlbyname("hw.acpi.battery.life", &capacity, &size_capacity, NULL,0) != 0) {
```

Comment:
Hmm, [top(1)](https://man.freebsd.org/top/1) seems to check `hw.acpi.battery.units` before `hw.acpi.battery.life`. https://github.com/freebsd/freebsd-src/commit/fc8ae86a8f25

Reply:
Thanks.
I added a test and throw an error if there is no battery units.